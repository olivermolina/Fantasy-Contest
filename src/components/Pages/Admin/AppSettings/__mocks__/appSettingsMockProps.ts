export const appSettingsMockProps = {
  appSettings: {
    BONUS_CREDIT_FREE_ENTRY_EQUIVALENT: '25',
    STAKE_TYPE_FREE_ENTRY: 'ALL_IN,INSURED',
    NUMBER_OF_PLAYERS_FREE_ENTRY: '2,3,4',
    MAX_MATCH_DEPOSIT_AMOUNT: '100',
    DEPOSIT_AMOUNT_OPTIONS: '10,25,50,75,100,200',
    RELOAD_BONUS_TYPE: 'FLAT',
    RELOAD_BONUS_AMOUNT: '0',
    MAX_RETENTION_BONUS: '50',
    RETENTION_BONUS_WEEKLY_CHANCE: '1',
    RETENTION_BONUS_MATCH_MULTIPLIER: '1',
    REFERRAL_CREDIT_AMOUNT: '25',
    REFERRAL_CUSTOM_TEXT:
      'Refer a friend from the link or code in your profile and get $25 in bonus credit!!',
    MIN_BET_AMOUNT: '1',
    MAX_BET_AMOUNT: '50',
    MAX_MARKET_ODDS: '150',
    MIN_MARKET_ODDS: '-150',
    initial: false,
    CHALLENGE_PROMO_MESSAGE:
      'Pick 2-4 players. Predict if they will get MORE or LESS than their projection.',
    REPEAT_ENTRIES_LIMIT: '1',
  },
  onSubmit: () => console.log('onSubmit'),
};
