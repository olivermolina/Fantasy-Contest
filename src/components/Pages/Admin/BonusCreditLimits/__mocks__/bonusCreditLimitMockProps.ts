import { BetStakeType } from '@prisma/client';

export const bonusCreditLimitMockProps = {
  onSubmit: () => console.log('submit'),
  data: {
    signupFreeEntry: false,
    bonusCreditFreeEntryEquivalent: 50,
    bonusCreditLimits: [
      {
        contestCategoryId: '1',
        numberOfPicks: 2,
        id: '30387acd-660c-46f0-a9e6-af87c007e6fb',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 25,
      },
      {
        contestCategoryId: '2',
        numberOfPicks: 3,
        id: '316555e5-b447-468a-b57f-381bbf64919a',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 50,
      },
      {
        contestCategoryId: '3',
        numberOfPicks: 4,
        id: 'fc063fcd-1a25-46e2-9361-d947eec1f9a9',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 40,
      },
      {
        contestCategoryId: '4',
        numberOfPicks: 5,
        id: 'a2542740-e90f-4763-8a3a-f7afba68898c',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 20,
      },
    ],
  },
  openAddUsersFreeEntryConfirmDialog: () => console.log('open'),
};
