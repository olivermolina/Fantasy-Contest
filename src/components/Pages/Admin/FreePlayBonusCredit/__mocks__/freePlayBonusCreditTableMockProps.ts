import { PaymentStatusCode } from '~/constants/PaymentStatusCode';
import { TransactionType } from '@prisma/client';

export const freePlayBonusCreditTableMockProps = {
  onSubmitDeletePlayFreeCredit: () => console.log('delete'),
  data: [
    {
      id: '1',
      transactionDate: new Date(),
      amountBonus: 100,
      status: PaymentStatusCode.COMPLETE,
      type: TransactionType.CREDIT,
    },
    {
      id: '2',
      transactionDate: new Date(),
      amountBonus: 100,
      status: PaymentStatusCode.CANCELLED,
      type: TransactionType.DEBIT,
    },
  ],
  totalBonusCreditBalance: 0,
};
