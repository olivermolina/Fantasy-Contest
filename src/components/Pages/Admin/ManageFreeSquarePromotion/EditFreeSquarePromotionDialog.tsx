import React, { useEffect } from 'react';
import {
  AppBar,
  Button,
  Dialog,
  FormControl,
  FormGroup,
  FormHelperText,
  FormLabel,
  IconButton,
  TextField,
  Toolbar,
  Typography,
} from '@mui/material';
import CloseIcon from '@mui/icons-material/Close';
import { ManageFreeSquarePromotionRowModel } from './ManageFreeSquarePromotion';
import { Controller, useForm } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers/yup';
import FormControlLabel from '@mui/material/FormControlLabel';
import Checkbox from '@mui/material/Checkbox';
import * as Yup from 'yup';
import { ContestCategory, League } from '@prisma/client';

export interface AddFreeSquarePromotionInput {
  /**
   * Free Square ID
   */
  id: string | null;
  /**
   * Market ID of the line
   */
  marketId: string;
  /**
   * Sel ID of the line
   */
  selId: number;
  /**
   * The discount to be applied to the line
   * @example 95
   */
  discount: number;
  /**
   * The list of contest pick category ID's
   * @example [pickId1, pickId2, pickId3]
   */
  pickCategories: [string, ...string[]];
  /**
   * Boolean to check if it can be used in a free entry
   */
  freeEntryEnabled: boolean;
  /**
   * Max stake/bet amount
   */
  maxStake: number;
  /**
   * League
   */
  league: League;
}

const InputValidationSchema = Yup.object().shape({
  discount: Yup.number()
    .typeError('Please provide a discount')
    .min(0, 'Minimum discount is 0')
    .max(100, 'Maximum discount is 100'),
  maxStake: Yup.number()
    .typeError('Please provide a max stake/bet.')
    .min(1, 'Minimum discount is 1'),
  pickCategories: Yup.array().min(1, 'Must have at least one category'),
});

interface EditFreeSquarePromotionDialogProps {
  /**
   * Close dialog callback function
   */
  handleClose: () => void;
  /**
   * Boolean to show dialog
   * @example true
   */
  open: boolean;
  /**
   * ManageFreeSquarePromotion row model
   */
  row?: ManageFreeSquarePromotionRowModel;
  /**
   * Available contest pick categories
   */
  contestCategories: ContestCategory[];
  /**
   * Callback function to save the free square promotion discount
   */
  handleSave: (formData: AddFreeSquarePromotionInput) => void;
}

/**
 * A dialog form for saving a free square details
 */
export default function EditFreeSquarePromotionDialog(
  props: EditFreeSquarePromotionDialogProps,
) {
  const { handleClose, open, row, contestCategories, handleSave } = props;
  const {
    formState: { errors },
    handleSubmit,
    control,
    register,
    reset,
  } = useForm<AddFreeSquarePromotionInput>({
    resolver: yupResolver(InputValidationSchema),
    defaultValues: {
      id: row?.freeSquare?.id,
      marketId: row?.id,
      selId: row?.selId,
      discount: row?.freeSquare?.discount,
      pickCategories:
        row?.freeSquare?.pickCategories.map(
          (pickCategory) => pickCategory.id,
        ) || [],
      freeEntryEnabled: row?.freeSquare?.freeEntryEnabled,
      maxStake: row?.freeSquare?.maxStake,
      league: row?.league as League,
    },
    mode: 'all',
    shouldFocusError: true,
    shouldUseNativeValidation: false,
  });

  const onSubmit = (formData: AddFreeSquarePromotionInput) => {
    handleSave(formData);
    handleClose();
  };

  useEffect(() => {
    reset({
      id: row?.freeSquare?.id,
      marketId: row?.id,
      selId: row?.selId,
      discount: row?.freeSquare?.discount,
      pickCategories:
        row?.freeSquare?.pickCategories.map(
          (pickCategory) => pickCategory.id,
        ) || [],
      freeEntryEnabled: row?.freeSquare?.freeEntryEnabled,
      maxStake: row?.freeSquare?.maxStake,
      league: row?.league as League,
    });
  }, [row]);

  return (
    <Dialog open={open} onClose={handleClose} fullWidth maxWidth={'xs'}>
      <AppBar sx={{ position: 'relative' }}>
        <Toolbar>
          <Typography sx={{ ml: 2, flex: 1 }} variant="h6" component="div">
            Free Square Promotion
          </Typography>
          <IconButton
            edge="start"
            color="inherit"
            onClick={handleClose}
            aria-label="close"
          >
            <CloseIcon />
          </IconButton>
        </Toolbar>
      </AppBar>

      <form onSubmit={handleSubmit(onSubmit)}>
        <div className={'flex flex-col justify-center gap-2 p-4'}>
          <p>
            Name: <span className={'font-bold'}>{row?.playerName} </span>
          </p>
          <p>
            Stat: <span className={'font-bold'}>{row?.statName} </span>
          </p>
          <p>
            Total: <span className={'font-bold'}>{row?.total}</span>
          </p>
          <div className={'flex flex-col items-baseline text-lg gap-1'}>
            <p>Discount:</p>
            <TextField
              hiddenLabel
              type={'number'}
              variant="standard"
              fullWidth
              {...register('discount')}
              error={!!errors?.discount}
              helperText={errors?.discount?.message}
              size={'small'}
              inputProps={{
                step: 0.01,
              }}
              margin={'none'}
            />
          </div>

          <div className={'flex flex-col items-baseline text-lg gap-1'}>
            <p>Max Stake/Bet:</p>
            <TextField
              hiddenLabel
              type={'number'}
              variant="standard"
              fullWidth
              {...register('maxStake')}
              error={!!errors?.maxStake}
              helperText={errors?.maxStake?.message}
              size={'small'}
              inputProps={{
                step: 0.01,
              }}
              margin={'none'}
            />
          </div>

          <FormControl component="fieldset" error={!!errors?.pickCategories}>
            <FormLabel component="legend">Select # of players:</FormLabel>
            <FormGroup row>
              <Controller
                name="pickCategories"
                control={control}
                render={({ field }) => (
                  <>
                    {contestCategories.map((category) => (
                      <FormControlLabel
                        {...field}
                        key={category.id}
                        label={`${category.numberOfPicks} Picks`}
                        control={
                          <Checkbox
                            defaultChecked={
                              !!row?.freeSquare?.pickCategories.find(
                                (item) => item.id === category.id,
                              )
                            }
                            onChange={() => {
                              const contestCategoryId = category.id;
                              const values: string[] = field.value;

                              if (!values) {
                                field.onChange([contestCategoryId]);
                                return;
                              }

                              if (!values?.includes(contestCategoryId)) {
                                field.onChange([
                                  ...field.value,
                                  contestCategoryId,
                                ]);
                                return;
                              }

                              const newCategories = field.value.filter(
                                (category) => category !== contestCategoryId,
                              );
                              field.onChange(newCategories);
                            }}
                          />
                        }
                      />
                    ))}
                  </>
                )}
              />
            </FormGroup>
            <FormHelperText>{errors?.pickCategories?.message}</FormHelperText>
          </FormControl>

          <Controller
            name="freeEntryEnabled"
            control={control}
            render={({ field }) => (
              <FormControlLabel
                {...field}
                label={'Can use bonus credit'}
                control={
                  <Checkbox
                    checked={field.value}
                    onChange={(e) => field.onChange(e.target.checked)}
                  />
                }
              />
            )}
          />
          <Button variant={'contained'} type="submit">
            Save
          </Button>
        </div>
      </form>
    </Dialog>
  );
}
