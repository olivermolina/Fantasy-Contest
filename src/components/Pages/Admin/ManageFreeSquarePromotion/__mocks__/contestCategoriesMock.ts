import { ContestCategory } from '@prisma/client';
import { faker } from '@faker-js/faker';

export const contestCategoriesMock: ContestCategory[] = [
  {
    id: faker.datatype.uuid(),
    numberOfPicks: 2,
    allInPayoutMultiplier: 3,
    primaryInsuredPayoutMultiplier: 2,
    secondaryInsuredPayoutMultiplier: 0.5,
  },
  {
    id: faker.datatype.uuid(),
    numberOfPicks: 3,
    allInPayoutMultiplier: 5,
    primaryInsuredPayoutMultiplier: 2.5,
    secondaryInsuredPayoutMultiplier: 1.25,
  },
  {
    id: faker.datatype.uuid(),
    numberOfPicks: 4,
    allInPayoutMultiplier: 10,
    primaryInsuredPayoutMultiplier: 5,
    secondaryInsuredPayoutMultiplier: 1.25,
  },
];
