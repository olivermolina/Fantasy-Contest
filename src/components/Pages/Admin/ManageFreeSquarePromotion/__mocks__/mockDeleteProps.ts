export const mockDeleteProps = {
  open: true,
  setOpen: () => console.log('open'),
  freeSquareId: '123',
  handleDelete: () => console.log('delete'),
};
