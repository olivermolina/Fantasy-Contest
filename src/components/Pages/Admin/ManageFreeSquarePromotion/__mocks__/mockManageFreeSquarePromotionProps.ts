import { League } from '@prisma/client';
import { contestCategoriesMock } from './contestCategoriesMock';

export const mockManageFreeSquarePromotionProps = {
  data: [
    {
      id: '63182-83-5227',
      selId: 5227,
      statName: 'Points',
      total: 11.5,
      matchName: 'STAN @ ORE ST',
      playerName: 'Glenn Taylor Jr.',
      marketId: '63182-83-5227',
      league: 'NCAAB',
      matchTime: 'Mar 02, 2023 @ 09:00 PM',
      freeSquare: {
        id: '1',
        discount: 95,
        pickCategories: [
          {
            id: '2 pick',
            numberOfPicks: 2,
          },
          {
            id: '3 pick',
            numberOfPicks: 3,
          },
          {
            id: '4 pick',
            numberOfPicks: 4,
          },
        ],
        maxStake: 25,
        freeEntryEnabled: false,
      },
      freeSquareDiscount: 95,
      discountedTotal: 1,
      picksCategory: '2,3,4',
      maxStake: 25,
      freeEntryEnabled: false,
    },
    {
      id: '63194-86-3541',
      selId: 3541,
      statName: 'Points Rebounds and Assists',
      total: 16.5,
      matchName: 'ARIZ @ USC',
      playerName: 'Kerr Kriisa',
      league: 'NCAAB',
      matchTime: 'Mar 02, 2023 @ 11:00 PM',
      freeSquare: {
        id: '1',
        discount: 98,
        pickCategories: [
          {
            id: '2 pick',
            numberOfPicks: 2,
          },
          {
            id: '3 pick',
            numberOfPicks: 3,
          },
          {
            id: '4 pick',
            numberOfPicks: 4,
          },
        ],
        maxStake: 25,
        freeEntryEnabled: false,
      },
      freeSquareDiscount: 95,
      discountedTotal: 1,
      picksCategory: '2,3,4',
      maxStake: 25,
      freeEntryEnabled: false,
    },
  ],
  setSelectedLeague: () => alert('setSelectedLeague'),
  handleSave: () => console.log('save'),
  contestCategories: contestCategoriesMock,
  selectedLeague: League.NBA,
  handleDelete: () => console.log('delete'),
};
