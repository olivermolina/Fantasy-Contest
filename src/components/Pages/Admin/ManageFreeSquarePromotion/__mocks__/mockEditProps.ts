import { mockManageFreeSquarePromotionProps } from './mockManageFreeSquarePromotionProps';
import { contestCategoriesMock } from './contestCategoriesMock';

export const mockEditProps = {
  row: mockManageFreeSquarePromotionProps.data[0],
  open: true,
  handleClose: () => console.log('close'),
  contestCategories: contestCategoriesMock,
  handleSave: () => console.log('save'),
};
