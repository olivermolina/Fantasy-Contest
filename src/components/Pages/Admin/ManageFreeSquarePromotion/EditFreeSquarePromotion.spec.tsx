import { render } from '@testing-library/react';
import EditFreeSquarePromotionDialog from './EditFreeSquarePromotionDialog';
import { mockEditProps } from '~/components/Pages/Admin/ManageFreeSquarePromotion/__mocks__/mockEditProps';

describe('EditFreeSquarePromotionDialog', () => {
  it('should render correctly', () => {
    const { container } = render(
      <EditFreeSquarePromotionDialog {...mockEditProps} />,
    );
    expect(container).toMatchSnapshot();
  });
});
