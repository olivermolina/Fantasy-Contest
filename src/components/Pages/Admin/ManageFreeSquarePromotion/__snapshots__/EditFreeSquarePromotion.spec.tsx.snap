// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditFreeSquarePromotionDialog should render correctly 1`] = `
<div
  aria-hidden="true"
/>
`;
